assets/blocks/quiz/quiz-block/quiz-settings-answer-feedback.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * WordPress dependencies
+ */
+import { BaseControl, ToggleControl } from '@wordpress/components';
+import { useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Answer feedback settings for the quiz block.
+ *
+ * @param {Object}   props
+ * @param {Object}   props.options       Quiz options.
+ * @param {Function} props.setAttributes
+ */
+const QuizAnswerFeedbackSettings = ( { options, setAttributes } ) => {
+	let {
+		failedShowAnswerFeedback = null,
+		failedShowCorrectAnswers = null,
+		failedIndicateIncorrect = null,
+		allowRetakes,
+	} = options;
+
+	const hasAnyOptionSet =
+		null !== failedShowAnswerFeedback ||
+		null !== failedShowCorrectAnswers ||
+		null !== failedIndicateIncorrect;
+
+	failedShowAnswerFeedback = failedShowAnswerFeedback ?? ! allowRetakes;
+	failedShowCorrectAnswers = failedShowCorrectAnswers ?? ! allowRetakes;
+	failedIndicateIncorrect = failedIndicateIncorrect ?? ! allowRetakes;
+
+	/**
+	 * Set default values based on 'Allow Retakes' if nothing was set.
+	 */
+	useEffect( () => {
+		if ( ! hasAnyOptionSet ) {
+			setAttributes( {
+				options: {
+					...options,
+					failedShowAnswerFeedback,
+					failedShowCorrectAnswers,
+					failedIndicateIncorrect,
+				},
+			} );
+		}
+	}, [
+		hasAnyOptionSet,
+		setAttributes,
+		failedShowAnswerFeedback,
+		options,
+		failedShowCorrectAnswers,
+		failedIndicateIncorrect,
+	] );
+
+	/**
+	 * Set selected option.
+	 *
+	 * @param {string} optionKey
+	 */
+	const createChangeHandler = ( optionKey ) => ( value ) =>
+		setAttributes( {
+			options: {
+				...options,
+				[ optionKey ]: value,
+			},
+		} );
+
+	return (
+		<div>
+			<BaseControl
```

Comment:
Yeah, went with `BaseControl` for this to avoid having to re-create the help styles in core — since those are now defined there as styled components but not exported directly.
Updated to `h3` and added `id` in d32e500

Reply:
I think we could use kebab case for the `id` since it's used to write in the HTML.

Reply:
Updated in https://github.com/Automattic/sensei/pull/4344/commits/055fe91a7b34723275badca89d89266504c88376